app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -16,20 +31,83 @@ def save_ems_container_inventory(ems, hashes, target = nil)
     ems.save!
   end
 
-  def save_container_projects_inventory(ems, hashes, target = nil)
-    return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.container_projects.reset
-    deletes = if target.kind_of?(ExtManagementSystem)
-                :use_association
-              else
-                []
-              end
+  def initialize_inventory_collections(ems)
+    @inv_collections = {}
+    @inv_collections[:container_projects] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerProject,
+      :parent => ems,
+      :builder_params => {:ems_id => ems.id},
+      :association => :container_projects
+    )
+    @inv_collections[:container_quotas] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerQuota,
+      :parent => ems,
+      :builder_params => {:ems_id => ems.id},
+      :association => :container_quotas,
+      #:arel => ContainerQuota.joins(:container_project).where(:container_projects => {:ems_id => ems.id}),
+    )
+    @inv_collections[:container_quota_items] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerQuotaItem,
+      :parent => ems,
+      :association => :container_quota_items,
+      #:arel => ContainerQuotaItem.joins(:container_quota => :container_project).where(:container_projects => {:ems_id => ems.id}),
+      :manager_ref => [:container_quota, :resource],
+    )
+    @inv_collections[:container_nodes] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerNode,
+      :parent => ems,
+      :builder_params => {:ems_id => ems.id},
+      :association => :container_nodes,
+    )
+    # ContainerCondition is polymorphic child of ContainerNode & ContainerGroup
+    @inv_collections[:container_node_conditions] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerCondition,
+      :parent => ems,
+      :association => :container_node_conditions,
+      #:arel => ContainerCondition.joins(
+      #  'INNER JOIN "container_nodes" ON "container_conditions"."container_entity_id" = "container_nodes"."id"'
+      #).where(:container_entity_type => 'ContainerNode',
+      #        :container_nodes => {:ems_id => ems.id}),
+      :manager_ref => [:container_entity, :name],
+    )
+
+    # polymorphic child of ContainerNode & ContainerImage
+    # TODO these are dumb, find way to share same InventoryCollections via both.
```

Comment:
@Ladas @agrare can you help me here?
can I express ComputerSystem (and its has_one Hardware, has_one OperatingSystem) via one InventoryCollection, with polymorphic `managed_entity` link to different parents?
I see how to share code when I'll build the inventory objects, what I miss is how would I set `parent, association` / `arel` on the collection.
I need a joins trail back to the EMS, to be able to delete "my" objects but not "other's" objects, right?

(I could also live with 2x duplication here, but not with 15x for labels & tags)